Eatery/Controllers/Eateries/CollegetownEateriesViewController.swift

```diff
@@ -0,0 +1,190 @@
+    }
+
+    private func queryCollegetownEateries() {
+        NetworkManager.shared.getCollegetownEateries { [weak self] (eateries, error) in
+            guard let `self` = self else {
+                return
+            }
+
+            guard let eateries = eateries else {
+                if let error = error {
+                    self.updateState(.failedToLoad(error), animated: true)
+                }
+
+                return
+            }
+
+            self.allEateries = eateries
+            self.updateState(.presenting, animated: true)
+        }
+    }
+
+    private func showMenu(of eatery: CollegetownEatery) {
+        let menuViewController = CollegetownEateriesMenuViewController(eatery: eatery, delegate: self, userLocation: userLocation)
+        navigationController?.pushViewController(menuViewController, animated: true)
+    }
+
+}
+
+// MARK: -
+
+extension CollegetownEateriesViewController: EateriesViewControllerDataSource {
+
+    func eateriesViewController(_ evc: EateriesViewController, eateriesToPresentWithSearchText searchText: String, filters: Set<Filter>) -> [Eatery] {
+        guard let eateries = allEateries else {
+            return []
+        }
+
+        var filteredEateries = eateries
+
+        if !searchText.isEmpty {
+            filteredEateries = filter(eateries: filteredEateries, withSearchText: searchText)
+        }
+
+        filteredEateries = filter(eateries: filteredEateries, withFilters: filters)
+
+        return filteredEateries
+    }
+
+    private func filter(eateries: [CollegetownEatery], withSearchText searchText: String) -> [CollegetownEatery] {
+        return eateries.filter { eatery in
+            if search(searchText, matches: eatery.name) {
+                return true
+            }
+
+            if let activeEvent = eatery.activeEvent(atExactly: Date()),
+                activeEvent.menu.stringRepresentation.flatMap({ $0.1 }).contains(where: { search(searchText, matches: $0) }) {
+                return true
+            }
+
+            return false
+        }
+    }
+
+    private func filter(eateries: [CollegetownEatery], withFilters filters: Set<Filter>) -> [CollegetownEatery] {
+        var filteredEateries = eateries
+
+        filteredEateries = filteredEateries.filter {
+            if filters.contains(.swipes) { return $0.paymentMethods.contains(.swipes) }
+            if filters.contains(.brb) { return $0.paymentMethods.contains(.brb) }
+            return true
+        }
+
+        let selectedCategoryFilters = filters.intersection(categoryFilters).map { $0.rawValue }
+        if !selectedCategoryFilters.isEmpty {
+            filteredEateries = filteredEateries.filter { eatery -> Bool in
+                eatery.categories.contains { category -> Bool in
+                    selectedCategoryFilters.contains { categoryFilter -> Bool in
+                        search(category, matches: categoryFilter) || search(categoryFilter, matches: category)
```

Comment:
Shouldn't searching if category matches categoryFilter or vice versa be the same?

I also feel like a comment here would be useful since it took a bit of time to understand (because of all the nested .contains) that you are checking an eateries' categories and then checking to see if that category matches with one of your selected filters.

Reply:
Adding a comment would be helpful.

But no, those would not be the same since `search(category, matches: categoryFilter)` finds occurrences of `category` in `categoryFilter` and `search(categoryFilter, matches: category)` does the opposite. 

I have no idea if this would make any practical difference, but it feels safer.